lib/jellyfish/room.ex

```diff
@@ -87,10 +100,18 @@ defmodule Jellyfish.Room do
   def init(max_peers) do
     state = new(max_peers)
     Logger.metadata(room_id: state.id)
+    {:ok, _pid} = Registry.register(Jellyfish.RoomRegistry, state.id, state.id)
+    Logger.info("Initialize room")
 
     {:ok, state}
   end
 
+  @impl true
+  def terminate(reason, state) do
+    :ok = Registry.unregister(Jellyfish.RoomRegistry, state.id)
+    reason
+  end
+
```

Comment:
It doesn't seem necessary.

Following up on this comment:

> (...) its keys are automatically removed from the registry but the change may not propagate immediately. This means certain operations may return processes that are already dead.
> (...)
> 
> Based on this information I think that it is possible to get the dead process from the registry, (...).

I've checked the implementation of `Registry`, and the removal of the registered process is based on a link (`register` calls `Process.link` with a partition pid and that process traps exits), so unregistration is only a matter of handling `:EXIT` message and removing entry from ETS in Registry process. Again, this doesn't fix a race condition, it may slightly reduce the chance of hitting it, so I'd get rid of this code